index.js

```diff
@@ -69,9 +68,14 @@ class ClinicBubbleprof extends events.EventEmitter {
       env: Object.assign({}, process.env, customEnv)
     })
 
-    if (this.detectPort) {
-      proc.stdio[3].once('data', data => this.emit('port', Number(data), proc, () => proc.stdio[3].destroy()))
-    }
+    proc.stdio[3].once('data', data => {
+      if (this.detectPort) {
```

Comment:
we should make sure that these features work together: if a port number is received, emit the 'port' event, and if the source warning value is received, emit the 'warning' event. Right now, the source warning will be treated as a port number if you're also using --detect-port.